app/src/main/java/com/github/Jenjamin3000/bootcamp/MapsFragment.java

```diff
@@ -0,0 +1,167 @@
+package com.github.Jenjamin3000.bootcamp;
+
+import static androidx.core.app.ActivityCompat.requestPermissions;
+
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+
+import android.Manifest;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.github.Jenjamin3000.bootcamp.utils.PermissionUtils;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.Polygon;
+import com.google.android.gms.maps.model.PolygonOptions;
+
+public class MapsFragment extends Fragment{
+
+    private GoogleMap map;
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
+    /**
+     * Flag indicating whether a requested permission has been denied
+     */
+    private boolean permissionDenied = false;
+    private OnMapReadyCallback callback = googleMap -> {
+        map = googleMap;
+
+        enableMyLocation();
+
+        //Listener when user clicks on the "my position" button
+        map.setOnMyLocationButtonClickListener(() -> {
+            Toast.makeText(getActivity(), "MyLocation button clicked", Toast.LENGTH_SHORT).show();
+            return false;
+        });
+
+        //Listener when user clicks on the "my position" blue dot
+        map.setOnMyLocationClickListener(location -> {
+            Toast.makeText(getActivity(), "Current location:\n" + location, Toast.LENGTH_LONG).show();
```

Comment:
I know this is just for demo purposes, but watch out when trying to print a location, the text it spits out seems bugged to me; maybe print the longitude and latitude individually?

Reply:
Yes it's because its uses a _fused location provider_ but I haven't had the time to look further into. For now I guess it's ok...